code/notebook/src/cleaning.py

```diff
@@ -0,0 +1,92 @@
+import pandas as pd
+import re
+import nltk
+from nltk.corpus import stopwords
+from nltk.stem import PorterStemmer, WordNetLemmatizer
+from nltk.tokenize import word_tokenize
+from nltk import pos_tag
+nltk.download('punkt')
+nltk.download('stopwords')
+nltk.download('wordnet')
+nltk.download('averaged_perceptron_tagger')
+
+stemmer = PorterStemmer()
+lemmatizer = WordNetLemmatizer()
+
+def nlp_cleaning(text):
+    """
+    Perform various text cleaning operations on the input text.
+
+    Parameters:
+    text (str): The input text to be cleaned.
+
+    Returns:
+    str: The cleaned text after applying the following operations:
+         
+    - Convert the text to lowercase.
+    - Remove URLs from the text.
+    - Remove non-ASCII characters.
+    - Remove numbers.
+    - Remove punctuation.
+    - Remove extra whitespaces.
+    """
+    if not isinstance(text, str):
```

Comment:
Can we do the following preprocessing on the episode column.

I have seen that the data has a lot of missing values but there is a pattern to it. 
Movies are usually just individual pieces of media and don't usually have episodes hence in that case, we do not have the episode count populated.

Can we do imputation on the records where we can find "movie" in the title, the value should be filled as 1

Reply:
After this, we can populate the rest of the missing values as -1